How to extend grid with Javascript extensions (#240)
PrestaShop/docs: src/content/1.7/development/components/grid/tutorials/extend-grid-with-javascript.md

weight: 4
---

> # Extend a grid with Javascript extensions. Preparation

Reviewer: `Preparation` ?

> Javascript grid extensions supplements grid functionality and enhances user experience.

Reviewer: ```suggestion
Javascript grid extensions supplement grid functionality and enhance user experience.
```

For example, most grids contains SortingExtension which provides sorting by columns functionality, 
BulkActionCheckboxExtension which enables interactive selection of bulk actions and so on.

> In this tutorial we use an example of taxes grid to show how to use extensions.

Reviewer: ```suggestion
In this tutorial we use a grid that display taxes items to show how to use extensions.
```

To start we will need to have a plain grid already created in php side
(you can find more docs about grid here https://devdocs.prestashop.com/1.7/development/components/grid).

> After following some guides we should be ready. Our grid looks like this:

Reviewer: ```suggestion
After following the Grid documentation directions you should now have a fully working php grid. The twig template should look like this:
```
